packages/server/modules/webhooks/services/webhooks.ts

```diff
@@ -0,0 +1,183 @@
+import { getServerInfo as getServerInfoFn } from '@/modules/core/services/generic'
+import { ForbiddenError } from '@/modules/shared/errors'
+import {
+  CountWebhooksByStreamId,
+  CreateWebhook,
+  CreateWebhookEvent,
+  DeleteWebhook,
+  GetWebhookById,
+  UpdateWebhook
+} from '@/modules/webhooks/domain/operations'
+import { Webhook } from '@/modules/webhooks/domain/types'
+import { SetValuesNullable } from '@speckle/shared'
+import crs from 'crypto-random-string'
+import {
+  StreamWithOptionalRole,
+  getStream as getStreamFn
+} from '@/modules/core/repositories/streams'
+import {
+  getUser as getUserFn,
+  UserWithOptionalRole
+} from '@/modules/core/repositories/users'
+import { Knex } from 'knex'
+
+const MAX_STREAM_WEBHOOKS = 100
+
+export const createWebhook =
+  ({
+    createWebhookConfig,
+    countWebhooksByStreamId
```

Comment:
Kinda subjective so you don't have to change this, but I tend to avoid destructuring the deps here and leave it in a `deps` object, because:

1) Less busy function signature. Repeating the same dependency twice - first as an actual destructuring operation and then to specify it's time - looks busy/mesy IMO.
2) If you don't destructure the deps and invoke the dep from that object (e.g. `deps.invokeX()`) you can always be sure that you're in fact invoking the injected dependency, not a function with the same name from the outer scope, in case you forgot to destructure it. This is relevant when we have those `typeof X` injections for old operations that aren't yet DI supported. If you do `getUser: typeof getUser` and forget to destructure it from `deps`, you will actually use the one imported in the module not the injected version. I've already accidentally made this mistake in the past.

Reply:
I think, at some point, we should agree on one way or the other.
But for now, I'll leave it there since, as you said, it is subjective.